src/engine/renderer/tr_shade.cpp

```diff
@@ -1580,8 +1594,17 @@ static void Render_forwardLighting_DBS_proj( shaderStage_t *pStage,
 	GL_CheckErrors();
 
 	// bind u_DiffuseMap
-	GL_BindToTMU( 0, pStage->bundle[ TB_DIFFUSEMAP ].image[ 0 ] );
-	gl_forwardLightingShader_projXYZ->SetUniform_TextureMatrix( tess.svars.texMatrices[ TB_DIFFUSEMAP ] );
+	if ( pStage->type == stageType_t::ST_LIGHTMAP )
+	{
+		// standalone lightmap stage: paint shadows over a white texture
+		GL_BindToTMU( 0, tr.whiteImage );
+	}
+	else
+	{
+		GL_BindToTMU( 0, pStage->bundle[ TB_DIFFUSEMAP ].image[ 0 ] );
+
+		gl_forwardLightingShader_projXYZ->SetUniform_TextureMatrix( tess.svars.texMatrices[ TB_DIFFUSEMAP ] );
+	}
```

Comment:
Should this be a helper function? Sounds like this pattern might need to be repeated for other types of situations?

Reply:
Right, it can be a helper function. We may even do `pStage->bundle[ TB_DIFFUSEMAP ].image[ 0 ] = tr.whiteImage` in `tr_shader.cpp` at shader processing time to simplify the code here.

Reply:
I've set the image in `tr_shader.cpp`.

Now making a helper function for a `if` would probably be overkill. It's also possible that `if` is just to skip useless uniform setting as a fullwhite image renders the same whatever the coordinates, and the code may now work without any `if` (I have not tried).